src/JsonSerialisation.ts

```diff
@@ -48,16 +41,214 @@ export class JsonSerialisation{
 		return list;
 	}
 
-	public static DictionaryRuntimeObjsToJObject(dictionary: Map<string, InkObject>){
-		let jsonObj: JObject = {};
-
+	public static WriteDictionaryRuntimeObjs(writer: SimpleJson.Writer, dictionary: Map<string, InkObject>){
+		writer.WriteObjectStart();
 		for (let [key, value] of dictionary){
-			let runtimeObj = asOrNull(value, InkObject);
-			if (runtimeObj != null)
-				jsonObj[key] = this.RuntimeObjectToJToken(runtimeObj);
+			writer.WritePropertyStart(key);
+			this.WriteRuntimeObject(writer, value);
+			writer.WritePropertyEnd();
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteListRuntimeObjs(writer: SimpleJson.Writer, list: InkObject[]){
+		writer.WriteArrayStart();
+		for (let value of list){
+			this.WriteRuntimeObject(writer, value);
+		}
+		writer.WriteArrayEnd();
+	}
+
+	public static WriteIntDictionary(writer: SimpleJson.Writer, dict: Map<string, number>){
+		writer.WriteObjectStart();
+		for (let [key, value] of dict){
+			writer.WriteIntProperty(key, value);
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteRuntimeObject(writer: SimpleJson.Writer, obj: InkObject): void{
+		// var container = obj as Container;
+		let container = asOrNull(obj, Container);
+		if (container) {
+			return this.WriteRuntimeContainer(writer, container);
```

Comment:
Nit, but imo this should be 

```
this.WriteRuntimeContainer(writer, container);
return;
```

Reply:
Haha, wasn't done on purpose. I agree, I'll make the change.